solution.js

```diff
@@ -1,5 +1,157 @@
 const { nums, words } = require("./data/data.js");
 
+
+class Node {
+  constructor(data) {
+      this.data = data;
+      this.next = null; 
+  }
+};
+
+
+class LinkedList {
+  constructor(head = null) {
+      this.head = head
+  };
+
+  size() {
+      let count = 0;
+      let currentNode = this.head;
+      while(currentNode) {
+          count++;
+          currentNode = currentNode.next;
+      }
+      return count; 
+  };
+
+  search(value) {
+      let currentNode = this.head;
+      while(currentNode) {
+          if (currentNode.data === value) {
+              return currentNode;
+          };
+          currentNode = currentNode.next;
+      };
+  };
+
+  clear() {
+      this.head = null; 
+  }
+
+  getFirst() {
+      return this.head;
+  }
+
+  getLast() {
+      let currentNode = this.head
+      while(currentNode.next != null){
+      currentNode = currentNode.next;
+      }
+      return currentNode;
+  }
+
+  insert(value) {
+      let newNode = new Node(value);
+      if(!this.head){
+          this.head = newNode;
+      } else {
+          newNode.next = this.head;
+          this.head = newNode;
+      }
+  }
+
+
+  delete(position) {
+      if (position <= 0 || position > this.size()) {
+          return;
+        }
+        if (position === 1) {
+          this.head = this.head.next;
+          return;
+        }
+        let currentNode = this.head;
+        let count = 1;
+        while (count < position - 1) {
+          currentNode = currentNode.next;
+          count++;
+        }
+        currentNode.next = currentNode.next.next;
+  };
+
+  isEmpty() {
+    let count = 0;
+    let currentNode = this.head;
+    while(currentNode) {
+        count++;
+        currentNode = currentNode.next;
+    }
+    if(count === 0){
+      return true
+    } else {
+      return false
+    }
+  }
+
+  getKth(position) {
+    let count = 0;
+    let currentNode = this.head;
+    while(currentNode != null) {
+      count ++
+      if(count === position){
+        return currentNode
+      }
+      currentNode = currentNode.next
+    }
+  }
+
+  getKthToLast(kth) {
+    let slow = this.head;
+    let fast = this.head.next;
+    for (let i = 0; i < kth; i++) {
+      fast = fast.next
+    }
+    while (fast != null) {
+      slow = slow.next;
+      fast = fast.next;
+    }
+    return slow;
+  }
+
+  toArray() {
+    let currentNode = this.head
+    let arr = []
+    while(currentNode) {
+      arr.push(currentNode.data)
+      currentNode = currentNode.next
+    }
+    return arr
+  }
+
+  containsDuplicates() {
+    let visited = new Set();
```

Comment:
Using a set you could compare the length of the set to the size of the linked list for shorter, cleaner code